src/tools/compiletest/src/header.rs

```diff
@@ -892,85 +882,54 @@ fn iter_header(
         // with a warm page cache. Maybe with a cold one.
         let original_line = &ln;
         let ln = ln.trim();
-        if ln.starts_with("fn") || ln.starts_with("mod") {
-            return;
```

Comment:
This previously skipped any `//@` after seeing the first occurrence of `fn` or `mod`, meaning that the mixed up `//@` invalid FileCheck directives in that assembly test was never caught. This might be however load-bearing for performance, but I'll need to benchmark it.

Reply:
I would lean towards retaining the status quo in this PR, and maybe pursuing a separate follow-up PR for trying to remove the early-stop behaviour.

Reply:
Fair enough, I'll keep that early-return.